lib/runners/processor/actionlint.rb

```diff
@@ -0,0 +1,110 @@
+module Runners
+  class Processor::Actionlint < Processor
+    SCHEMA = _ = StrongJSON.new do
+      extend Schema::ConfigTypes
+
+      # @type self: SchemaClass
+      let :config, base(
+        target: target,
+
+        # Native options
+        ignore: one_or_more_strings?,
+      )
+
+      let :issue, object(
+        shellcheck_issue: object?(id: string, severity: string, line: number, column: number, message: string, url: string),
+      )
+    end
+
+    ALLOWED_ERRORS = [
+      "no project was found in any parent directories of",
+      "no YAML file was found in",
+    ].freeze
+    SHELLCHECK_MESSAGE_FORMAT = /shellcheck reported issue in this script: (?<id>SC[^:]+):(?<severity>\w+):(?<line>\d+):(?<column>\d+): (?<message>.*)$/.freeze
+
+    register_config_schema SCHEMA.config
+
+    def self.config_example
+      <<~'YAML'
+        root_dir: project/
+        target:
+          - .github/workflows/one_workflow.yml
+          - .github/workflows/another_workflow.yml
+        ignore:
+          - "job .* needs job"
+          - "unknown Webhook event"
+      YAML
+    end
+
+    # This analyser uses git metadata (.git/).
+    def use_git_metadata_dir?
+      true
+    end
+
+    def analyzer_options
+      [].tap do |opts|
+        opts << "-format" << "{{json .}}"
+        # Specify error message patterns to ignore
+        Array(config_linter[:ignore]).each { |v| opts << "-ignore" << v }
+        # After -- separator, specify target files to analyze
+        opts << "--"
+        Array(config_linter[:target]).each { |v| opts << v }
+      end
```

Comment:
[suggestion] One refactoring idea:

```suggestion
      [
        "-format", "{{json .}}",

        # Specify error message patterns to ignore
        *(Array(config_linter[:ignore]).flat_map { |v| ["-ignore", v] }),

        # After -- separator, specify target files to analyze
        "--", *Array(config_linter[:target]),
      ]
```

But unfortunately, `bundle exec rake steep:check'[true]'` is necessary to avoid a Steep error. The cause is Steep fails to infer the return value type of `Array#flat_map`.

Reply:
What can be done in these cases? Add a new expected error in `steep_expectations.yml`? 

AFAIK, the github steep check runs `bundle exec rake steep:check`, not `bundle exec rake steep:check'[true]'`.

Reply:
If you want to update `steep_expectations.yml`, please run `bundle exec rake steep:check'[true]'` 🙆🏼‍♂️

Reply:
Changed along with steep expectations.

Reply:
Thanks!